llvm/include/llvm/Support/KnownBits.h

```diff
@@ -487,6 +487,20 @@ struct KnownBits {
   /// Update known bits based on XORing with RHS.
   LLVM_ABI KnownBits &operator^=(const KnownBits &RHS);
 
+  /// Shift known bits left by ShAmt. Shift in bits are unknown.
+  KnownBits &operator<<=(unsigned ShAmt) {
```

Comment:
This breaks the convention that `KnownBits::operatorXX` corresponds to doing `operatorXX` on the underlying value -- if we followed that convention then this operator would shift in known zeros.

Reply:
&, |, ^ are the only operators with this property. Ideally they'd be named `KnownBits::and`, `KnownBits::or`, and `KnownBits::xor` like add, sub, mul, shl, etc. But C/C++ treats `and`, `or`, `xor` as keywords.

Reply:
> Ideally

Well, that's an opinion... Personally I like the ergonomics of `operator&` etc and would have liked to use it in more cases like `operator+` instead of `add`, but that didn't work out because those functions needed extra arguments.

Anyway please don't consider my comments as blocking. I don't really object to the patch.